Store.js

```diff
@@ -140,14 +150,109 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
+						// affect the callback to the observe function if the item is observable
+						Observable.observe(this.store[i], this.observeCallback.bind(this));
+					}
+					collection = processQueryResult.call(this, this.store.filter(this._isQueried, this));
+					// affect the callback to the observe function if the array is observable
+					this._observeResult = ObservableArray.observe(this.store, this.observeCallback.bind(this));
 				}
+				return this.processCollection(collection);
 			} else {
 				this.initItems([]);
 			}
 		},
 
 		/**
+		 * Synchronously deliver change records of the Observe function for the Array
+		 * Do NOT deliver change records of the Observe function for Objects
+		 */
+		deliver: dcl.superCall(function (sup) {
+			return function () {
+				sup.call();
+				if (this._observeResult !== null && this._observeResult !== undefined) {
+					this._observeResult.deliver();
+				}
+			};
+		}),
+
+		/**
+		 * Called to verify if the item respect the query conditions
+		 * @param item
+		 * @protected
+		 */
+		_isQueried: function (item) {
+			for (var prop in this.query) {
+				if (item[prop] !== this.query[prop]) {
+					return false;
+				}
+			}
+			return true;
+		},
+
+		/**
+		 * Called when a modification is done on the array or its items.
+		 * @param changeRecords - send by the Observe function
+		 */
+		observeCallback: function (changeRecords) {
+			for (var i = 0; i < changeRecords.length; i++) {
+				// array modified
+				if (Array.isArray(changeRecords[i].object) === true) {
+					this.observeCallbackArray(changeRecords[i]);
+				// one item of the array modified
+				} else {
+					this.observeCallbackItems(changeRecords[i]);
+				}
+			}
+		},
+
+		/**
+		 * Called when a modification is done on the array.
+		 * @param changeRecords - send by the Observe function
+		 */
+		observeCallbackArray: function (change) {
+			if (change.type === "splice") {
+				var j;
+				for (j = 0; j < change.removed.length; j++) {
+					var evtRemoved = {previousIndex: change.index};
+					this._itemRemoved(evtRemoved);
+				}
+				for (j = 0; j < change.addedCount; j++) {
+					var evtAdded = {
+						index: change.index + j,
+						target: this.store[change.index + j]
+					};
+					if (this.renderItems !== null && this.renderItems !== undefined) {
+						evtAdded.index = evtAdded.index <= this.renderItems.length ?
+							evtAdded.index : this.renderItems.length;
+					}
+					// affect the callback to the observe function if the item is observable
+					Observable.observe(this.store[change.index + j], this.observeCallback.bind(this));
+					if (this._isQueried(evtAdded.target, this.query)) {
+						this._itemAdded(evtAdded);
+					}
+				}
+			}
+		},
+
+		/**
+		 * Called when a modification is done on the items.
+		 * @param changeRecords - send by the Observe function
+		 */
+		observeCallbackItems: function (change) {
+			if (change.type === "add" || change.type === "update" || change.type === "delete") {
```

Comment:
I guess the `this.observeCallback()` (if called from [this line](https://github.com/ibm-js/delite/pull/410/files#diff-3306122b521ab15fecaabf6fdc73a3dbR232)) can group change records by target object, and call `this._itemUpdated()` just once for each object.